src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java

```diff
@@ -597,8 +597,7 @@ Peer getPeerWithId(short id) throws SignalingException {
         if (id == SALTYRTC_ADDR_SERVER) {
             return this.server;
         } else if (this.isResponderId(id)) {
-            //noinspection ConstantConditions
-            if (this.getState() == SignalingState.TASK && this.responder != null & this.responder.getId() == id) {
+            if (this.getState() == SignalingState.TASK && this.responder != null && this.responder.getId() == id) {
```

Comment:
Note here the short-circuit evaluation. I was not sure why there was a single `&` before?

Reply:
With a single `&` a NPE is possible.

Reply:
Rephrasing: Quite sure what @dbrgn did there was unintended and wrong.